src/SpaceTime/IBackedBy.cs

```diff
@@ -0,0 +1,17 @@
+﻿
+namespace Bearded.Utilities.SpaceTime
+{
+    /// <summary>
+    /// Represents a type that is backed by a specific singular numeric type value.
+    /// </summary>
+    /// <typeparam name="T">The type of the backing value.</typeparam>
+    public interface IBackedBy<out T>
+        where T : struct
+    {
+        /// <summary>
+        /// Returns the numeric value of the type.
+        /// </summary>
+        T NumericValue { get; }
+    }
+
+}
```

Comment:
The idea is to allow for automatic generation of vertex attribute arrays from types that implement this interface, without the generator having to know all those types. This would allow users to define Position2 vectors, etc. in their vertex data, and have it be parsed correctly into a vec2 in GLSL for example. I can see the argument that maybe this should not be inside THIS library however... thoughts?